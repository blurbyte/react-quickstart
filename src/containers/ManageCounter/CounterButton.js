import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';


import Button from 'components/Button';

const StyledButton = styled(Button)`
  margin-right: 1.4rem;

  &:last-child {
    margin-right: 0;
  }
`;

const CounterButton = ({ disabled, children, onClick }) => (
  <StyledButton onClick={onClick} disabled={disabled}>
    {children}
  </StyledButton>
);

CounterButton.propTypes = {
  disabled: PropTypes.bool,
  children: PropTypes.node,
  onClick: PropTypes.func
};

export default CounterButton;
