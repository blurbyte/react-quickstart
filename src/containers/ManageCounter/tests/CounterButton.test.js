import React from 'react';
import renderer from 'react-test-renderer';

import CounterButton from '../CounterButton';

describe('<CounterButton />', () => {
  it('should render correctly', () => {
    // component init
    const component = renderer.create(
      <CounterButton />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
