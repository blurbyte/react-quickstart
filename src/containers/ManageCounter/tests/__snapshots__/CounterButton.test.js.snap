// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterButton /> should render correctly 1`] = `
<button
  className="khyxLt"
  disabled={undefined}
  onClick={undefined}
/>
`;
